functionary/functionary/settings/base/__init__.py

```diff
@@ -148,6 +153,7 @@
 AUTHENTICATION_BACKENDS = [
     "django.contrib.auth.backends.ModelBackend",
     "core.auth.backends.CoreBackend",
+    "allauth.account.auth_backends.AuthenticationBackend",
```

Comment:
Could we instead just pre-configure several authentication providers, and then allow the admin user of the Functionary instance to toggle their enabled state? For example, we setup the basic configuration for Github, Gitlab, or Google out of the box. Then provide some admin interface to dynamically adjust the settings for those providers during runtime, or toggle whether or not the provider is enabled.

I am partly influenced by how [Pocketbase](https://pocketbase.io/) allow their admins to dynamically toggle and configure authentication providers during runtime, versus having to enable a provider prior to runtime: 
![image](https://user-images.githubusercontent.com/113923579/209407771-9c631ea1-2742-47f0-9eec-6002a3eb29a5.png)
![image](https://user-images.githubusercontent.com/113923579/209407801-4bba3a79-61d8-497b-9574-00b466e4f24d.png)

This isn't something that should be done in this PR, but just an idea to throw out there.

Reply:
What you are describing here is effectively what the plan is and we've already got a follow-on ticket for the administration of the providers.  See #154.